python/ray/dag/tests/test_accelerated_dag.py

```diff
@@ -176,6 +192,32 @@ def f(x):
         dag.experimental_compile()
 
 
+@pytest.mark.parametrize("num_actors", [1, 4])
```

Comment:
Does this scenario work? Can you also add a test? 

1. run dag that has user exception in the middle
2. begin_read and raise an exception
3. reuse the dag (without calling end_read from the previous result).

Reply:
That definitely won't work, it's not something supported right now.

Reply:
I see. So rn we should make sure to use try/finally for all dag (for vllm pr)